kong/plugins/apifortress/fortress_serializer.lua

```diff
@@ -0,0 +1,117 @@
+-- The serializer. Builds a data structure that is compatible with
+-- the API Fortress API expectations
+
+local stringy = require "stringy"
+
+local _M = {}
+
+-- pre-indexing global functions
+local str_startswith = stringy.startswith
+local str_endswith = stringy.endswith
+local str_sub = string.sub
+local str_len = string.len
+local table_insert = table.insert
+local encode_base64 = ngx.encode_base64
+
+function _M.serialize(ngx)
+
+  -- pre-indexing to reduce the number of lookups later on
+  local ngx_req = ngx.req
+  local ngx_resp = ngx.resp
+  local ngx_var = ngx.var
+  local ngx_ctx = ngx.ctx
```

Comment:
```lua
local ngx_req = ngx.req
local ngx_resp = ngx.resp
```

Those serve little purpose. Ideally, you would cache the functions used in `serialize` (that are stateless, not request-bound) on the main chunk level:

```
local req_get_method = ngx.req.get_method
local req_get_headers = ngx.req.get_headers
-- etc...
```